include/tuvx/linear_algebra/linear_algebra.hpp

```diff
@@ -31,13 +31,13 @@ namespace tuvx
   /// @brief Fill a std::vector with random values
   /// @param x Vector to allocate and fill
   template<typename T>
-  void FillRandom(std::vector<T> &x);
+  void FillRandom(std::vector<T> &x, const unsigned &seed);
 
   /// @fn Initialize random matrix function
   /// @brief Fills a matrix with uniformly distributed random values.
   /// @param A tridiagonal matrix to allocate and fill
   template<typename T>
-  void FillRandom(TridiagonalMatrix<T> &A);
+  void FillRandom(TridiagonalMatrix<T> &A, const unsigned &seed, bool diagonally_dominant = false);
```

Comment:
Can we update the documentation for these API calls as well?  `seed` is pretty self explanatory but `diagonally_dominant` might be confusing for folks who haven't had linear algebra in a while and isn't clear if it's a feature of the matrix being passed in or an action the API will take.  It might also be a good idea to change this to `make_diagonally_dominant` to further clarify that this is an action the `FillRandom` call will have to perform.

Reply:
Good point. I will make this change in my next commit.